compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1410,6 +1410,33 @@ object SymDenotations {
     def namedType(using Context): NamedType =
       if (isType) typeRef else termRef
 
+    /** The typeRef where `pre.O$.this` is changed to `pre.O.type` if `O` is a non-static object
```

Comment:
These documentation comments need to be updated since we also do this for static objects now.

Reply:
Done.